### PR TITLE
5007 set operations

### PR DESCRIPTION
**Pull request by leviathan747**

### CODE REVIEW

File: src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+| ==          | equality                         | any                                        |
+| !=          | inequality                       | any                                        |
+| <           | less than                        | integer, real, date, timestamp, string     |
+| >           | greater than                     | integer, real, date, timestamp, string     |
+| <=          | less than or equal to            | integer, real, date, timestamp, string     |
+| >=          | greater than or equal to         | integer, real, date, timestamp, string     |
+| and         | logical conjunction              | boolean                                    |
+| or          | logical disjunction              | boolean                                    |
+| not         | boolean negation                 | boolean                                    |
+| \*empty     | instance reference/set empty     | instance reference, instance reference set |
```
Comment (keithbrown) [inline src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md]: I think this and the next should be replaced with empty<sup>1</sup> and then number the reference at the end of the table.  The asterisk he could be confused as part of the name of the operator.

File: src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+| \*not_empty | instance reference/set not empty | instance reference, instance reference set |
+
+\* see section ["Additional unary operators"](#additional_unary_operators)
```
Comment (keithbrown) [inline src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md]: numer as 1

File: src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+### Note
+
+* Both operands of equality and comparison operators must be of the same type or
```
Comment (keithbrown) [inline src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md]: Partially true.  BP provides preference for "Allow promotion of integer to real" and "Allow lossy assignment of real to integer" that play in here.

Comment (leviathan747) [reply src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md]: hmm. does that apply? doesn't that only apply to assignments?

File: src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md (hunk 1 above)
Comment (leviathan747) [reply src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md]: done

File: src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md (hunk 2 above)
Comment (leviathan747) [reply src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md]: done

File: src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md (hunk 3 above)
Comment (leviathan747) [reply src/org.xtuml.bp.doc/Reference/OAL/Expressions/Expressions.md]: i just checked. you were right. i added a note and reference to the section of the docs that talks about those preferences.

**Pull request merged**